include/common/oclengine.hpp

```diff
@@ -183,8 +202,20 @@ class OCLEngine {
     int GetDefaultDeviceID() { return default_device_context->context_id; }
     /// Pick a default device, for QEngineOCL instances that don't specify a preferred device.
     void SetDefaultDeviceContext(DeviceContextPtr dcp);
+    /// Initialize the OCL environment, with the option to save the generated binaries. Binaries will be saved/loaded from the folder path "home". This returns a Qrack::OCLInitResult object which should be passed to SetDeviceContextPtrVector().
+    static OCLInitResult InitOCL(bool buildFromSource = false, bool saveBinaries = false, std::string home = "*");
+    /// Get default location for precompiled binaries:
+    static std::string GetDefaultBinaryPath()
+    {
+#if defined(_WIN32) && !defined(__CYGWIN__)
```

Comment:
Pro-tip - check for the existence of QRACK_OCL_PATH as an override for both of these, so it can be set from the shell auto-magically.

Reply:
I can do that. Obviously, this is a compile-time solution, but the user has other options at run-time, as well.

Reply:
Done.

Reply:
No, I mean using getenv.  See if there's an environment variable set.  No need to plumb this through cmake.